javalib/src/main/scala/java/util/AbstractMap.scala

```diff
@@ -0,0 +1,199 @@
+package java.util
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+
+object AbstractMap {
+
+  private def entryEquals[K, V](entry: Map.Entry[K, V], other: Any): Boolean = {
+    other match {
+      case other: Map.Entry[_, _] =>
+        entry.getKey == other.getKey && entry.getValue == other.getValue
```

Comment:
Ah, my bad, here. Working on something else today made me realize that `==` is not equivalent to `equals` when numbers and characters are concerned.

I recommend defining a helper function
```scala
def sameValue(x: AnyRef, y: AnyRef) =
  if (x eq null) y eq null
  else x.equals(y)
```
and use that to compare keys and values. This function might be useful for other comparisons when implementing the JDK collections.